### ISSUE TITLE
Add ability to pass Beam PipelineOptions to pipelines and InteractiveContext

### ISSUE THREAD

**Issue by codesue** [tensorflow/tfx#2332]
Currently, it's only possible to pass pipeline options to pipelines and InteractiveContext as a list of arguments (`beam_pipeline_args`). If pipeline options are defined in a `apache_beam.options.pipeline_options.PipelineOptions` object, they must first be converted into a list of arguments before they can be passed to a pipeline.

Adding the ability to pass PipelineOptions directly would improve usability and make it easier to reuse code.

**Comment by casassg** [comment]
Yup, this would make development quite more easy to compare than when using Beam Python SDK

**Comment by codesue** [comment]
A specific example of how this would be an improvement is not having to to convert the option `{'use_public_ips': False}` to a flag. There's a detailed explanation on [this Beam JIRA ticket](https://issues.apache.org/jira/browse/BEAM-9093?focusedCommentId=17191064&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17191064).

**Comment by singhniraj08** [comment]
@codesue,

**Comment by singhniraj08** [comment]
@codesue,

Are you still looking for a resolution? We are planning on prioritising the issues based on the community interests. Please let us know if this issue still persists with the latest TFX 1.13 release so that we can work on fixing it. Thank you for your contributions.

**Comment by codesue** [comment]
Thanks for following up, @singhniraj08! This issue is no longer relevant to me, so I'll go ahead and close it.
